metrics-operator/controllers/common/providers/prometheus/prometheus.go

```diff
@@ -112,6 +114,19 @@ func (r *KeptnPrometheusProvider) EvaluateQueryForStep(ctx context.Context, metr
 	return getResultForStepMatrix(result)
 }
 
+func (r *KeptnPrometheusProvider) setupApi(ctx context.Context, provider metricsapi.KeptnMetricsProvider, getter RountripGetter) (prometheus.API, error) {
```

Comment:
let's move the getter to `KeptnPrometheusProvider` struct

Reply:
I want to mock this and I think it is a bit of an overkill to add an interface, thus injected here